engine/src/witness/common/chunked_chain_source/latest_then.rs

```diff
@@ -19,6 +20,137 @@ impl<Inner, F> LatestThen<Inner, F> {
 		Self { inner, f }
 	}
 }
+
+fn latest_then_stream<'a, ItemStream, Fut, ThenFn, Output, Index, Hash, Data, Info, HistoricInfo>(
```

Comment:
Would be nice to have a function description

Reply:
A better place would be on the `pub` adapter function itself, as a doc comment. `engine/src/witness/common/chunked_chain_source/chunked_by_time/builder.rs` . Perhaps we still include a code comment here to help describe the "how"